flutter_searchbox_ui/lib/src/components/selected_filters.dart

```diff
@@ -0,0 +1,363 @@
+  ///      return '$id: $value';
+  ///    },
+  ///   ...
+  /// )
+  /// ```
+  final String Function(String id, String value)? filterLabel;
+
+  /// If `true` then display `Clear All` button. Defaults to `true`.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   showClearAll: false,
+  ///   ...
+  /// )
+  /// ```
+  final bool? showClearAll;
+
+  /// To modify the clear all label.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   clearAllLabel: "Vanish All",
+  ///   ...
+  /// )
+  /// ```
+  final String? clearAllLabel;
+
+  /// A map of active widgets' default values.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   defaultValues: () {
+  ///         "price-filter": { start: 10000 }
+  ///     },
+  ///   ...
+  /// )
+  /// ```
+  final Map<String, dynamic>? defaultValues;
+
+  /// If `true` then `Clear All` action would reset the filter values to default
+  /// values. User must define the `defaultValues` prop when `resetToDefault` is `true`.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   resetToDefault: true,
+  ///   defaultValues: () {
+  ///         "price-filter": { start: 10000 }
+  ///     },
+  ///   ...
+  /// )
+  /// ```
+  final bool? resetToDefault;
+
+  /// If `true` then SelectedFilters will not show default values of active widgets.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   hideDefaultValues: true,
+  ///
+  ///   ...
+  /// )
+  /// ```
+  final bool? hideDefaultValues;
+
+  /// Callback function, triggered when all filters are cleared.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   onClearAll: () {
+  ///         // do something here
+  ///     },
+  ///   ...
+  /// )
+  /// ```
+  final void Function()? onClearAll;
+
+  /// Callback function, triggered when a specific filter is cleared.
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   onClear: (id, value) {
+  ///         // do something here
+  ///     },
+  ///   ...
+  /// )
+  /// ```
+  final void Function(String, dynamic)? onClear;
+
+  /// It is used for custom rendering the filters
+  /// For example,
+  /// ```dart
+  /// SelectedFilters(
+  ///   ...
+  ///   buildFilters: (options) {
+  ///     List<Widget> widgets = [];
+  ///      options!.selectedValues.forEach((id, filterValue) {
+  ///      widgets.add(
+  ///        Chip(
+  ///          label: Text(
+  ///              ' $id --- ${options.getValueAsString(filterValue)}'),
+  ///          onDeleted: () {
+  ///            options.clearValue(id);
+  ///          },
+  ///        ),
+  ///      );
+  ///    });
+  ///    return Wrap(
+  ///      spacing: 16.0,
+  ///     crossAxisAlignment: WrapCrossAlignment.start,
+  ///     children: widgets,
+  ///    );
+  ///   },
+  ///   ...
+  /// )
+  /// ```
+
+  final Widget Function([BuildOptions? options])? buildFilters;
```

Comment:
`options` has to be used, so shouldn't be an optional param.

Reply:
fixed

Reply:
`final Widget Function(BuildOptions options)? buildFilters;`